src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,27 +1,38 @@
 import cx from "classnames";
 import { useCallback, useState, useRef } from "react";
+import Wrapper from "./Wrapper";
 import { IS_TOUCH } from "config/env";
 
 const OPEN_DELAY = 0;
 const CLOSE_DELAY = 100;
 
-export default function Tooltip(props) {
+type Props = {
+  handle: React.ReactNode;
+  renderContent: () => React.ReactNode;
+  position?: string;
+  trigger?: string;
+  className?: string;
+  disableHandleStyle?: boolean;
+  handleClassName?: string;
+  onDisabled?: boolean;
+};
+
+export default function Tooltip(props: Props) {
   const [visible, setVisible] = useState(false);
-  const intervalCloseRef = useRef(null);
-  const intervalOpenRef = useRef(null);
+  const intervalCloseRef = useRef<null | number>(null);
+  const intervalOpenRef = useRef<null | number>(null);
 
   const position = props.position ?? "left-bottom";
   const trigger = props.trigger ?? "hover";
 
   const onMouseEnter = useCallback(() => {
     if (trigger !== "hover" || IS_TOUCH) return;
-
     if (intervalCloseRef.current) {
       clearInterval(intervalCloseRef.current);
       intervalCloseRef.current = null;
     }
     if (!intervalOpenRef.current) {
-      intervalOpenRef.current = setTimeout(() => {
+      intervalOpenRef.current = window.setTimeout(() => {
```

Comment:
hmm why we need window here?

if typescript shows an error here it's strange because in other places we use just setTimeout (see lib/sleep.ts)

Reply:
The difference is assigning to a variable with type null | number. I will update type.